### ISSUE TITLE
Callbacks for initializePayment are mixed. getting the SUCCESS data inside failureCallback().

### ISSUE THREAD

**Issue by UnSensei** [nitish24p/react-native-upi#6]
The library works fine and does as it advertises, but the returns are mixed. successcallback doesnt return anything.

**Comment by nitish24p** [comment]
Can you tell me for which case this happens?

**Comment by UnSensei** [comment]
hey, im not really sure i understood what you meant by which case, but i tried it with google Pay and Amazon Pay, the payments are going through to the right owner, but the  transactionID and SUCCESS CASE which are meant to be returned through the successCallback are instead returned in the failureCallback. [Medium Article by sombody](https://medium.com/@nonstablei/calling-upi-from-react-native-app-411efc4da53a) this is a link to medium page where the author has shown what i mean.

**Comment by nitish24p** [comment]
Aahh okay got it. will fix this over the weekend

**Comment by UnSensei** [comment]
@nitish24p Thank you, this will help me clean my code to standard.

**Comment by nitish24p** [comment]
Yes was travelling last weekend. was away from work.. will do this

**Comment by prasanthown** [comment]
i m also faced same problem
Fix the response comparison code on

> node_modules/react-native-upi-payment/android/src/main/java/com/upi/payment/UpiPaymentModule.java

_on line no 89_

`if (bundle.getString("response") == "SUCCESS") {`

Replace with

`if (data.getStringExtra("Status").trim().equals("SUCCESS")){`

**Comment by nitish24p** [comment]
@prasanthown can you give a PR.. i'll merge it and push to npm
